mapreduce/src/main/java/org/infinispan/hadoop/mapreduce/InfinispanInputFormat.java

```diff
@@ -0,0 +1,51 @@
+package org.infinispan.hadoop.mapreduce;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.hadoop.mapreduce.impl.ConvertingRecordReader;
+import org.infinispan.hadoop.mapreduce.impl.DefaultSplitter;
+import org.infinispan.hadoop.mapreduce.impl.InfinispanCache;
+import org.infinispan.hadoop.mapreduce.impl.InfinispanRecordReader;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * InputFormat for Infinispan remote caches, with best data locality.
+ *
+ * @author gustavonalle
+ * @since 0.1
+ */
+public class InfinispanInputFormat<K, V> extends InputFormat<K, V> {
+
+   @Override
+   public List<InputSplit> getSplits(JobContext jobContext) throws IOException, InterruptedException {
+      Configuration configuration = jobContext.getConfiguration();
+      InfinispanConfiguration infinispanConfiguration = new InfinispanConfiguration(configuration);
+      InfinispanCache<K, V> inputCache = InfinispanCache.getInputCache(infinispanConfiguration);
+      return new DefaultSplitter().calculateSplits(inputCache.getCacheTopology());
```

Comment:
eventually, we can put set InfinispanSplitter in the configuration. it may be allow to do implementations based on the cache configuration (distributed or replicated)